### PR TITLE
Add the ProvisioningRequest's classname annotation

### LINKED ISSUE #2051: Add the ProvisioningRequest's classname annotation to pods
**What happened**:
Pods created by Kueue that use ProvisioningRequest lack the `cluster-autoscaler.kubernetes.io/provisioning-class-name` annotation

**What you expected to happen**:
Pods created by Kueue that use ProvisioningRequest to have the `cluster-autoscaler.kubernetes.io/provisioning-class-name` annotation

**How to reproduce it (as minimally and precisely as possible)**:
Create a Job that uses a ProvisioningRequest

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #2051 

#### Special notes for your reviewer:
I've tested it e2e manually

#### Does this PR introduce a user-facing change?

```release-note
Pods created by Kueue have now the ProvisioningRequest's classname annotation
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
@mimowo @alculquicondor Do we need to cherry pick this to release v0.6 branch?
I'm not sure if we need to do it since IIRC, the provisioning requests is not yet cutting a release (no objections).

File: pkg/controller/admissionchecks/provisioning/constants.go [hunk 1]
```diff
@@ -17,9 +17,10 @@ limitations under the License.
 package provisioning
 
 const (
-	ConfigKind            = "ProvisioningRequestConfig"
-	ControllerName        = "kueue.x-k8s.io/provisioning-request"
-	ConsumesAnnotationKey = "cluster-autoscaler.kubernetes.io/consume-provisioning-request"
+	ConfigKind             = "ProvisioningRequestConfig"
+	ControllerName         = "kueue.x-k8s.io/provisioning-request"
+	ConsumesAnnotationKey  = "cluster-autoscaler.kubernetes.io/consume-provisioning-request"
+	ClassNameAnnotationKey = "cluster-autoscaler.kubernetes.io/provisioning-class-name"
```
[the remark this answers is not in the document]
Comment (yaroslava-serdiuk) [reply pkg/controller/admissionchecks/provisioning/constants.go:23]: Will move those annotation in autoscaler repo.

**Pull request merged**